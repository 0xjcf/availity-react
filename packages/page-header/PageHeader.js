import React from 'react';
import PropTypes from 'prop-types';
import Breadcrumbs from '@availity/breadcrumbs';
import AppIcon from '@availity/app-icon';
import Feedback from '@availity/feedback';
import PayerLogo from '@availity/payer-logo';

const PageHeader = ({
  payerId,
  spaceName,
  spaceId,
  children,
  appName,
  appAbbr,
  iconColor,
  branded,
  crumbs: ogCrumbs,
  feedback,
  component,
  ...props
}) => {
  let crumbs = ogCrumbs;
  if (spaceId || spaceName) {
    crumbs = [
      { name: spaceName, url: spaceId && `/web/spaces/spaces/#/${spaceId}` },
    ];
  }
  return (
    <React.Fragment>
      <div className="d-flex align-items-start">
        <Breadcrumbs crumbs={crumbs} active={appName || children} /> {component}
      </div>
      <h1 className="h4 page-header page-header-brand" {...props}>
        <div className="page-header-title">
          {!payerId && appAbbr && (
            <AppIcon color={iconColor} branded={branded} title={appName}>
              {appAbbr}
            </AppIcon>
          )}{' '}
          {children || appName}
        </div>
        {!payerId && feedback && (
          <Feedback
            appName={appName}
            className="float-md-right d-inline-block"
          />
        )}
        {payerId && (
          <PayerLogo
            spaceId={spaceId}
            payerId={payerId}
            className="float-md-right d-inline-block"
          />
        )}
      </h1>
    </React.Fragment>
  );
};

PageHeader.propTypes = {
  appName: PropTypes.string.isRequired,
  spaceName: PropTypes.string,
  spaceId: PropTypes.string,
  appAbbr: PropTypes.string,
  iconColor: PropTypes.string,
  branded: PropTypes.bool,
  payerId: PropTypes.string,
  component: PropTypes.element,
  feedback: PropTypes.bool,
  children: PropTypes.node,
  crumbs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
    })
  ),
};

export default PageHeader;
