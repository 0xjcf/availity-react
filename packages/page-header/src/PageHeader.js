import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Breadcrumbs from '@availity/breadcrumbs';
import AppIcon from '@availity/app-icon';
import Feedback from '@availity/feedback';
import Spaces, { SpacesLogo, useSpaces, useSpacesContext } from '@availity/spaces';
import '../styles.scss';

const PageHeader = ({
  payerId,
  logo,
  spaceName,
  spaceId,
  children,
  appName,
  appAbbr,
  iconColor,
  branded,
  showCrumbs,
  crumbs,
  feedback: showFeedback,
  feedbackProps,
  titleProps: { className: titleClassName, ...restTitleProps },
  renderRightContent: RenderRightContent,
  renderRightClassName,
  component,
  tag: Tag,
  clientId,
  iconSrc,
  iconAlt,
  homeUrl,
  linkTag,
  className,
  ...props
}) => {
  const spaces = useSpaces(spaceId, payerId);

  let firstSpaceForPayerID;
  let firstSpaceForSpaceID;
  if (!spaceId && payerId && spaces[0]) {
    firstSpaceForPayerID = spaces[0];
  } else {
    firstSpaceForSpaceID = spaces[0];
    firstSpaceForPayerID = spaces[1];
  }
  const { loading: spacesIsLoading } = useSpacesContext() || {};

  const _space = firstSpaceForSpaceID || firstSpaceForPayerID;

  let payerLogo = null;
  if (payerId || logo) {
    const logoAttrs = {
      spaceId,
      payerId,
      className: 'd-inline-flex',
      skeletonProps: {
        width: 180,
        height: '100%',
      },
    };
    payerLogo =
      firstSpaceForPayerID || (logo && firstSpaceForSpaceID) || spacesIsLoading ? (
        <SpacesLogo {...logoAttrs} />
      ) : (
        <Spaces
          spaceIds={spaceId ? [spaceId] : undefined}
          payerIds={payerId ? [payerId] : undefined}
          clientId={clientId}
        >
          <SpacesLogo {...logoAttrs} />
        </Spaces>
      );
  }

  const _spaceName = spaceName || (_space && _space.name);
  if ((spaceId || _spaceName) && !crumbs) {
    const url =
      firstSpaceForSpaceID && firstSpaceForSpaceID.link && firstSpaceForSpaceID.link.url
        ? firstSpaceForSpaceID.link.url
        : `/web/spc/spaces/#/${spaceId}`;
    crumbs = [{ name: _spaceName, url }];
  }

  const feedback = useMemo(() => <Feedback appName={appName} {...feedbackProps} />, [appName, feedbackProps]);

  return (
    <>
      <div className="page-header-above">
        {showCrumbs ? (
          <>
            {React.isValidElement(crumbs) ? (
              crumbs
            ) : (
              <Breadcrumbs crumbs={crumbs} active={appName || children} homeUrl={homeUrl} linkTag={linkTag} />
            )}
          </>
        ) : null}
        {component}
      </div>
      <div className={classNames('page-header', className)} data-testid="page-header" {...props}>
        <Tag
          className={classNames('page-header-title', titleClassName)}
          data-testid="page-header-title"
          {...restTitleProps}
        >
          {!payerId && appAbbr && (
            <AppIcon
              data-testid="page-header-app-icon"
              color={iconColor}
              branded={branded}
              title={appName}
              src={iconSrc}
              alt={iconAlt}
              aria-hidden
            >
              {appAbbr}
            </AppIcon>
          )}
          {children || <h1>{appName}</h1>}
        </Tag>
        {!RenderRightContent ? (
          <div className="page-header-right">
            {showFeedback && feedback}
            {payerLogo}
          </div>
        ) : (
          <RenderRightContent className={renderRightClassName} payerLogo={payerLogo} feedback={feedback} />
        )}
      </div>
    </>
  );
};

PageHeader.propTypes = {
  /** Name of the application. */
  appName: PropTypes.string.isRequired,
  /** The name of the class. */
  className: PropTypes.string,
  /** Payer Space Name of the space in which this page is under. */
  spaceName: PropTypes.string,
  /** Payer Space ID of the space in which this page is under. */
  spaceId: PropTypes.string,
  /** If provided and payerId is not provided, the app icon will appear. */
  appAbbr: PropTypes.string,
  /** Potential values: "black", "blue", "green", "orange", "red". Only used if the app icon should appear. Default: "black". */
  iconColor: PropTypes.string,
  /** Triggers the app icon's "branded" styles. Only used if the app icon should appear. */
  branded: PropTypes.bool,
  /** The ID of the payer the application is for. If provided, the payer logo appears and not the app icon. */
  payerId: PropTypes.string,
  /** Allow rendering of an optional component in the top right of the header. */
  component: PropTypes.element,
  /** If true, the feedback loop button appears. */
  feedback: PropTypes.bool,
  /** See Feedback. Props to send to <Feedback /> component */
  feedbackProps: PropTypes.shape({ ...Feedback.propTypes }),
  /** Additional props passed to the page-header-title. */
  titleProps: PropTypes.object,
  /** This value appears at the h1 of the page. Overrides the appName to allow for custom elements (such as a "beta" tag). Default: ${appName}. */
  children: PropTypes.node,
  /** Custom link tag for the links. Default: <a> */
  linkTag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  /** Flag whether or not to show the breadcrumbs. Default: true */
  showCrumbs: PropTypes.bool,
  /** Array of Objects containing name (String) and url (String) properties. The ancestor pages which get passed to the Breadcrumbs component. See Breadcrumbs */
  crumbs: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
      })
    ),
    PropTypes.node,
  ]),
  /** If true will fetch the payer logo using the spaceId or payerId. */
  logo: PropTypes.bool,
  /** The tag to render the page header as. Default: <h1>. */
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  /** Used to customize the contents of the right side of the page header where the feedback and payer logo get rendered. Accepts the rendered payerLogo and feedback as props */
  renderRightContent: PropTypes.func,
  /** Renders the name of the right side of the page header */
  renderRightClassName: PropTypes.string,
  /** Url for the Home route. Default: public/apps/dashboard. */
  homeUrl: PropTypes.string,
  /** Client ID to use in Spaces to fetch the payer's logo */
  clientId: PropTypes.string,
  /** Image source for <AppIcon /> to be used instead of appAbbr. */
  iconSrc: PropTypes.string,
  /** Required if iconSrc is used. Image alt property of <AppIcon /> */
  iconAlt: PropTypes.string,
};

PageHeader.defaultProps = {
  tag: 'div',
  // SpaceId and payerId are defaulted to null to prevent `useSpace` from
  // returning a space we may not want
  spaceId: null,
  payerId: null,
  homeUrl: '/public/apps/dashboard',
  titleProps: {},
  renderRightClassName: 'page-header-right',
  showCrumbs: true,
};

export default PageHeader;
