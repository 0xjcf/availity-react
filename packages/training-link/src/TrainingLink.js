import React from 'react';
import PropTypes from 'prop-types';

const shouldProbablyBeInUIKit = {
  paddingTop: '12px',
};

const linkStyles = { fontWeight: 'bold' };

const TrainingLink = ({ name, link }) => (
  <span style={shouldProbablyBeInUIKit} className="ml-auto">
    Need Help?{' '}
    <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
      Watch a demo
    </a>{' '}
    for {name}
  </span>
);

TrainingLink.propTypes = {
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default TrainingLink;
