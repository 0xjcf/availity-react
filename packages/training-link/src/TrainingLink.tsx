import React from 'react';
import type { CSSProperties } from 'react';

const shouldProbablyBeInUIKit: CSSProperties = { paddingTop: '12px' };
const linkStyles: CSSProperties = { fontWeight: 'bold' };

export type TrainingLinkProps = {
  /**Link to training video.*/
  link?: string;
  /**Name of your app to make the sentence complete.*/
  name?: string;
};

const TrainingLink = ({ name, link }: TrainingLinkProps): JSX.Element => (
  <>
    {name ? (
      <span style={shouldProbablyBeInUIKit} className="ml-auto">
        Need Help?{' '}
        <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
          Watch a demo <span className="sr-only">(opens in a new tab)</span>
        </a>{' '}
        for {name}
      </span>
    ) : (
      <span style={shouldProbablyBeInUIKit} className="ml-auto">
        Need Help?{' '}
        <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
          Learn More
        </a>
      </span>
    )}
  </>
);

export default TrainingLink;
