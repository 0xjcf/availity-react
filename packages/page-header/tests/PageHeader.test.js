import React from 'react';
import renderer from 'react-test-renderer';
import TrainingLink from '@availity/training-link';
import PageHeader from '..';

describe('PageHeader', () => {
  test('should render', () => {
    const component = renderer.create(<PageHeader appName="Payer Space" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render app icon', () => {
    const component = renderer.create(
      <PageHeader appName="Payer Space" appAbbr="PS" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render app icon color', () => {
    const component = renderer.create(
      <PageHeader appName="Payer Space" appAbbr="PS" iconColor="green" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render app icon color branded', () => {
    const component = renderer.create(
      <PageHeader
        appName="Payer Space"
        appAbbr="PS"
        iconColor="green"
        branded
      />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render feedback', () => {
    const component = renderer.create(
      <PageHeader appName="Payer Space" feedback />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render children', () => {
    const component = renderer.create(
      <PageHeader appName="Payer Space">
        <p>this is cool</p>
      </PageHeader>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  test('should render trainingLink', () => {
    const component = renderer.create(
      <PageHeader
        appName="Payer Space"
        component={
          <TrainingLink
            name="Appeals"
            link="https://www.youtube.com/watch?v=GgwE94KZJ7E"
          />
        }
      >
        <p>this is cool</p>
      </PageHeader>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
