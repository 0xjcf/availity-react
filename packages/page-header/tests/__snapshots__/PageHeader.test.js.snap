// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
       
      Payer Space
    </div>
  </h1>,
]
`;

exports[`PageHeader should render app icon 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
      <span
        className="app-icon app-icon-black"
        title="Payer Space"
      >
        PS
      </span>
       
      Payer Space
    </div>
  </h1>,
]
`;

exports[`PageHeader should render app icon color 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
      <span
        className="app-icon app-icon-green"
        title="Payer Space"
      >
        PS
      </span>
       
      Payer Space
    </div>
  </h1>,
]
`;

exports[`PageHeader should render app icon color branded 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
      <span
        className="app-icon app-icon-branded-green"
        title="Payer Space"
      >
        PS
        <span
          className="caret"
        />
      </span>
       
      Payer Space
    </div>
  </h1>,
]
`;

exports[`PageHeader should render children 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
       
      <p>
        this is cool
      </p>
    </div>
  </h1>,
]
`;

exports[`PageHeader should render feedback 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
       
      Payer Space
    </div>
    <div
      className="float-md-right d-inline-block hidden-print dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="btn btn-light"
        onClick={[Function]}
        type="button"
      >
        Give Feedback
      </button>
      <div
        aria-hidden={true}
        className="p-3 dropdown-menu dropdown-menu-right"
        role="menu"
        style={
          Object {
            "width": "400px",
          }
        }
        tabIndex="-1"
        x-placement={undefined}
      >
        <div
          className="text-center"
        >
          <p>
            Tell us what you think about Payer Space.
          </p>
          <div
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-light"
              onClick={[Function]}
              style={
                Object {
                  "fontSize": "1.4em",
                  "padding": ".2em .4em",
                }
              }
              type="button"
            >
              <span
                className="icon icon-smile"
              />
              <span
                className="sr-only"
              >
                Smiley face
              </span>
            </button>
            <button
              className="btn btn-light"
              onClick={[Function]}
              style={
                Object {
                  "fontSize": "1.4em",
                  "padding": ".2em .4em",
                }
              }
              type="button"
            >
              <span
                className="icon icon-meh"
              />
              <span
                className="sr-only"
              >
                Meh face
              </span>
            </button>
            <button
              className="btn btn-light"
              onClick={[Function]}
              style={
                Object {
                  "fontSize": "1.4em",
                  "padding": ".2em .4em",
                }
              }
              type="button"
            >
              <span
                className="icon icon-frown"
              />
              <span
                className="sr-only"
              >
                Frowny face
              </span>
            </button>
          </div>
          <div
            className="invalid-feedback"
          >
            Please select one of the faces above
          </div>
          <textarea
            aria-label="Please provide any feedback, requests, and/or defects."
            className="mt-3 form-control"
            onChange={[Function]}
            placeholder="Feedback? Requests? Defects? (optional)"
          />
          <div
            className="mt-3 text-right"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </h1>,
]
`;

exports[`PageHeader should render trainingLink 1`] = `
Array [
  <div
    className="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      className=""
    >
      <ol
        className="breadcrumb"
      >
        <li
          aria-current={undefined}
          className="breadcrumb-item"
        >
          <a
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
     
    <span
      className="ml-auto"
      style={
        Object {
          "paddingTop": "12px",
        }
      }
    >
      Need Help?
       
      <a
        href="https://www.youtube.com/watch?v=GgwE94KZJ7E"
        rel="noopener noreferrer"
        target="_blank"
      >
        Watch a demo
      </a>
       
      for 
      Appeals
    </span>
  </div>,
  <h1
    className="h4 page-header page-header-brand"
  >
    <div
      className="page-header-title"
    >
       
      <p>
        this is cool
      </p>
    </div>
  </h1>,
]
`;
